### PR TITLE
Stop the reaper TTL test asking floats for a guarantee they do not give

### PR DESCRIPTION
**Pull request by TianYuan-Liu**
`test_a_delivered_result_is_dropped_after_the_ttl` has failed CI twice on
unrelated pull requests (#162, #163) and passed on re-run both times. It is not
intermittent in the usual sense — no sleep, no thread, no randomness — so it was
twice written off as noise. **It is arithmetic.**

### The bug

`deliver_result()` stamps `time.monotonic()`. `reap_delivered()` tests
`now - delivered_at >= ttl`, and the test handed it `now = stamp + 600`.

On a fresh CI runner `time.monotonic()` is *seconds since boot* — a few hundred.
At that magnitude `stamp + 600` is not exactly representable, so
`(stamp + 600) - stamp` comes back as `599.99999999999989` and the boundary
inverts.

Measured over 200,000 stamps drawn from `[100, 3000)`: **9.5% fail.**

### Why it never reproduced locally — and why I got this wrong once already

A laptop that has been up for weeks has a monotonic clock in the millions, where
the subtraction is exact. On #162 I tested this theory at `time.time()`
magnitudes, saw 0 failures in 200,000, and rejected it as not fitting. The
magnitude was the whole point.

| stamps drawn from | old form fails |
|---|---|
| `[100, 3000)` — fresh CI runner | **9.5%** |
| `~1.7e9` — `time.time()`, or a long-lived laptop's monotonic | 0% |

### What is *not* changed

The production code. At the exact TTL boundary a difference of 1e-13 either way
is meaningless, and there `now` is a fresh clock reading rather than
`stamp + ttl`, so a result missed by one status poll is reaped by the next.
What was wrong is the test asking for `>=` at exactly the boundary with a stamp
that cannot answer it.

### The fix

The two boundary cases set `delivered_at` to an exactly representable `1000.0`,
keeping what they exist to pin — that the comparison is `>=` and not `>` — plus:

- a guard asserting `STAMP`'s offsets are exact, so changing it to a clock
  reading fails loudly instead of inverting one run in ten;
- a guard reproducing the rounding itself, from a literal fresh-runner-magnitude
  reading **found by search**. A hand-picked "messy-looking" number does not
  reproduce it (`1234.5678901234567` subtracts exactly), which is worth having
  in the suite rather than only in a commit message;
- a third case still exercising the real `time.monotonic()` stamp, comfortably
  past the TTL, so pinning the boundary with chosen numbers cannot hide a stamp
  that was never set at all.

### Verified

With `STAMP` set back to a reproducing value (`1773.5985509907462`) the original
flake reappears **and** both new guards fail. With `1000.0`, 29/29 pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: PaintomicsServer/src/tests/test_finished_job_result_survives_a_lost_response.py [hunk 1]
```diff
@@ -233,19 +233,93 @@ def setUp(self):
...
+        against the form that was flaking. What broke was a stamp with a messy
+        fractional part, which is the only kind `time.monotonic()` returns.
+
+        So this asserts the property the two boundary cases above rely on,
+        directly: change STAMP to a clock reading and this fails rather than
+        the boundary silently inverting one run in ten.
+        """
+        for offset in (599, 600):
+            with self.subTest(offset=offset):
+                self.assertEqual((self.STAMP + offset) - self.STAMP,
```
[the remark this answers is not in the document]
Comment (TianYuan-Liu) [reply PaintomicsServer/src/tests/test_finished_job_result_survives_a_lost_response.py:301]: You are right on both counts, and I merged this before reading the comment — that was my mistake. Fixed in #165.

Measured your first point over 200,000 readings drawn from [100, 3000): the old guard fails **19.0%** of the time against a clock reading, not 100%. So it was a second flaky test standing next to the one it was meant to guard, and the docstring stated it as a certainty while contrasting it with the probabilistic bug — strictly worse than saying nothing.

It now asserts a property of the stamp itself, as you suggested: integral and below 2**53. An integral value in that range is exactly representable, and so is that value plus any integer offset that stays in range, so no rounding is possible for any offset on any machine. That fails 100% against a clock reading.

Your second point I handled by moving the check rather than adding another: it lives inside `_deliverAt` now, so every boundary case is covered by construction. The residual gap you named — deleting `_deliverAt` and reading `delivered_at` directly, where STAMP stays 1000.0 and the guard still passes — is now stated in the docstring instead of papered over. Nothing in that file can see a helper that is bypassed entirely.
